packages/flutter/lib/src/material/color_scheme.dart

```diff
@@ -209,39 +418,113 @@ class ColorScheme with Diagnosticable {
   /// The color displayed most frequently across your app’s screens and components.
   final Color primary;
 
-  /// A darker version of the primary color.
-  final Color primaryVariant;
+  /// A color that's clearly legible when drawn on [primary].
+  ///
+  /// To ensure that an app is accessible, a contrast ratio of 4.5:1 for [primary]
+  /// and [onPrimary] is recommended. See
+  /// <https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-contrast.html>.
+  final Color onPrimary;
+
+  final Color? _primaryContainer;
+  /// A color used for elements needing less emphasis than [primary].
+  Color get primaryContainer => _primaryContainer ?? primary;
+
+  final Color? _onPrimaryContainer;
+  /// A color that's clearly legible when drawn on [primaryContainer].
+  ///
+  /// To ensure that an app is accessible, a contrast ratio of 4.5:1 for
+  /// [primaryContainer] and [onPrimaryContainer] is recommended. See
+  /// <https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-contrast.html>.
+  Color get onPrimaryContainer => _onPrimaryContainer ?? onPrimary;
 
-  /// An accent color that, when used sparingly, calls attention to parts
-  /// of your app.
+  /// An accent color used for less prominent components in the UI, such as
+  /// filter chips, while expanding the opportunity for color expression.
   final Color secondary;
 
-  /// A darker version of the secondary color.
-  final Color secondaryVariant;
+  /// A color that's clearly legible when drawn on [secondary].
+  ///
+  /// To ensure that an app is accessible, a contrast ratio of 4.5:1 for [secondary]
+  /// and [onSecondary] is recommended. See
+  /// <https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-contrast.html>.
+  final Color onSecondary;
 
-  /// The background color for widgets like [Card].
-  final Color surface;
+  final Color? _secondaryContainer;
+  /// A color used for elements needing less emphasis than [secondary].
+  Color get secondaryContainer => _secondaryContainer ?? secondary;
 
-  /// A color that typically appears behind scrollable content.
-  final Color background;
+  final Color? _onSecondaryContainer;
+  /// A color that's clearly legible when drawn on [secondaryContainer].
+  ///
+  /// To ensure that an app is accessible, a contrast ratio of 4.5:1 for
+  /// [secondaryContainer] and [onSecondaryContainer] is recommended. See
+  /// <https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-contrast.html>.
+  Color get onSecondaryContainer => _onSecondaryContainer ?? onSecondary;
+
+  final Color? _tertiary;
+  /// A color used as a contrasting accent that can be balance [primary]
```

Comment:
Nit: looks like a word is missing from this sentence: "can be balance"

Reply:
D'oh. Thanks.